stock_quant_manual_assign/wizard/assign_manual_quants.py

```diff
@@ -63,20 +71,28 @@ def default_get(self, fields):
         available_quants = self.env['stock.quant'].search([
             ('location_id', 'child_of', move.location_id.id),
             ('product_id', '=', move.product_id.id),
-            ('qty', '>', 0),
-            '|',
-            ('reservation_id', '=', False),
-            ('reservation_id', '=', move.id)
+            ('quantity', '>', 0)
         ])
-        quants_lines = [{
-            'quant': x.id,
-            'lot_id': x.lot_id.id,
-            'in_date': x.in_date,
-            'package_id': x.package_id.id,
-            'selected': x in move.reserved_quant_ids,
-            'qty': x.qty if x in move.reserved_quant_ids else 0,
-            'location_id': x.location_id.id,
-        } for x in available_quants]
+
+        quants_lines = []
+        for x in available_quants:
+            line = {}
+            line['quant'] = x.id
+            line['lot_id'] = x.lot_id.id
+            line['on_hand'] = x.quantity
+            line['package_id'] = x.package_id.id
+            line['selected'] = False
+            line['qty'] = 0
+            move_lines = move.move_line_ids.filtered(
+                lambda ml:
+                ml.location_id == x.location_id and ml.lot_id == x.lot_id)
```

Comment:
I think this is too restrictive (or I don't understand the flow) ? IMHO you have to add condition as you maybe want to match quants that don't have lot if move line has no lot.

`... and (ml.lot_id == False and x.lot_id = False or ml.lot_id == x.lot_id)`

Reply:
On move lines, you can have multiple locations if products come from children of move location.

Try to store product A in STOCK > Shelf 1 and product B in STOCK > Shelf 2. Create a picking from stock to out with product A and product B.

Moves are from stock to out and move lines are from children locations.

Reply:
these lines of code is getting the correct reserved quant/lot info (form previous action) and display to the wizard form.